Makefile

```diff
@@ -20,12 +20,43 @@ help:
 	@echo "run 'make install_cli' to build and install the command line tool builds"
 	@echo "run 'make test-something' to run cargo tests matching 'something'"
 
+# We use bash features in some of our Makefile shell scripting
 SHELL = /bin/bash
+
+# The Rust versions required for Holochain development, and the required cargo options are configured here:
 CORE_RUST_VERSION ?= nightly-2019-01-24
 TOOLS_RUST_VERSION ?= nightly-2019-01-24
-CARGO = RUSTFLAGS="-Z external-macro-backtrace -D warnings" RUST_BACKTRACE=1 rustup run $(CORE_RUST_VERSION) cargo $(CARGO_ARGS)
-CARGO_TOOLS = RUSTFLAGS="-Z external-macro-backtrace -D warnings" RUST_BACKTRACE=1 rustup run $(TOOLS_RUST_VERSION) cargo $(CARGO_ARGS)
-CARGO_TARPULIN_INSTALL = RUSTFLAGS="--cfg procmacro2_semver_exempt -D warnings" RUST_BACKTRACE=1 cargo $(CARGO_ARGS) +$(CORE_RUST_VERSION)
+CARGO = RUSTFLAGS="-Z external-macro-backtrace -D warnings" rustup run $(CORE_RUST_VERSION) cargo -Z config-profile $(CARGO_ARGS)
+CARGO_TOOLS = RUSTFLAGS="-Z external-macro-backtrace -D warnings" rustup run $(TOOLS_RUST_VERSION) cargo -Z config-profile $(CARGO_ARGS)
+CARGO_TARPULIN_INSTALL = RUSTFLAGS="--cfg procmacro2_semver_exempt -D warnings" cargo -Z config-profile $(CARGO_ARGS) +$(CORE_RUST_VERSION)
+
+# All rustup and cargo invocations executed (directly in this Makefile, or indirectly eg. via npm)
+# must include these environment variables.  If we'd like to see Rust back-traces:
+export RUST_BACKTRACE=1
+
+# There are 3 methods to obtain the libsodium encryption library required by holochain-rust,
+# selectable by setting/clearing various RUST_SODIUM_...  environment variables (see:
+# https://github.com/maidsafe/rust_sodium).  These selections are implemented and enforced in
+# rust_sodium-sys/build.rs
+
+# 1) System installed: select by setting RUST_SODIUM_LIB_DIR. We need to find the location of the
+# system's libsodium dynamic library: at least version 1.0.12 is required.  On Mac, `brew install
+# libsodium`.  On Ubuntu Bionic (libsodium 1.0.16), Consmic (libsodium 1.0.16) and Disco (libsodium
+# 1.0.17): `apt-get install libsodium-dev`.  On Debian Stretch (stable, libsodium 1.0.11 *to old*,
+# add `buster` to /etc/apt/sources...), or Debian Buster (testing, libsodium 1.0.17): `apt-get -t
+# buster -u install libsodium-dev`.  On other Linux distros, ensure at least libsodium 1.0.12+ is
+# installed.
+RUST_SODIUM_LIB=$(shell find /usr/local/lib /usr/lib /lib -name 'libsodium.so' -o -name 'libsodium.dylib' | head -1)
+export RUST_SODIUM_LIB_DIR=$(dir $(RUST_SODIUM_LIB))
```

Comment:
I don't have the context for the node-js neon binding problems... but part of the reason for choosing rust_sodium-sys in the first place was it defaulted to downloading / building a pinned version of libsodium on all target os-s (linux, windows, macOs)... so we wouldn't have to deal with lame stale os versions...

Not saying we shouldn't do this... just wish we could get it to work the other way.

Reply:
I'll take another run at getting the download/build option working on Debian. I'm not sure why it doesn't work properly -- it should. If something can't build on Debian, there's something seriously wrong...

Reply:
Please sync with @philipbeadle and @thedavidmeister about this. They have spent MANY hours working on the lib-sodium issue, lets' not duplicate work.